File: desktop-staff/reservation-winforms/DTO/reservation/WalkInRequest.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace reservation_winforms.DTO.reservation
{
    public class WalkInRequest
    {
        [JsonProperty("guestCount")]
        public int GuestCount { get; set; }

        [JsonProperty("customerName")]
        public string CustomerName { get; set; }

        [JsonProperty("customerPhone")]
        public string CustomerPhone { get; set; }

        [JsonProperty("startTime")]
        public DateTime? StartTime { get; set; }

        [JsonProperty("endTime")]
        public DateTime? EndTime { get; set; }

        [JsonProperty("tableId")]
        public List<long> TableId { get; set; }

        [JsonProperty("mergeTables")]
        public bool MergeTables { get; set; }

        [JsonProperty("note")]
        public string Note { get; set; }
    }
}

File: backend-api/reservation-api/src/main/java/com/hcmute/reservation/model/dto/reservation/WalkInRequest.java
package com.hcmute.reservation.model.dto.reservation;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class WalkInRequest {
    @NotNull @Min(1)
    private Integer guestCount;

    private String customerName;

    private String customerPhone;

    // null = auto-detect now
    private LocalDateTime startTime;

    // Thời gian kết thúc dự kiến do Frontend tính toán truyền xuống
    private LocalDateTime endTime;

    // Cho phép chỉ định bàn cụ thể (optional, null = auto-assign)
    private List<Long> tableId;

    // Ghép bàn: nếu 1 bàn không đủ
    private boolean mergeTables;

    private String note;
}
